task_2/join.go

```diff
@@ -0,0 +1,124 @@
+package task_2
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func Start() {
+
+	var (
+		numOfChannels = rand.Intn(12) + 1
+		channels      []chan interface{}
+	)
+
+	fmt.Println(fmt.Sprintf("creating %d channels..", numOfChannels))
+	for i := 0; i < numOfChannels; i++ {
+		channels = append(channels, make(chan interface{}))
```

Comment:
I assume that messages in terms of a specific channel are produced independently and with random intervals. In this case, we might have to store new messages that are not ready to been received by the join function. 

There are two possible ways to deal with it: 
1. Using buffered channels;
2. Using some kind of persistence storages.